### PR TITLE
LPS-139347 Delete and start again workflow tasks

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://issues.liferay.com/browse/LPS-139347

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
Just started reviewing :)

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -666,6 +672,26 @@ private void _createTable(
 		runSQL(dynamicObjectDefinitionTable.getCreateTableSQL());
 	}
 
+	private void _deleteObjectEntriesWorkflowTasks(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java]: Hey @mateussgn , Would you mind to change the name of this method to _deleteWorkflowInstanceLinks?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 2]
```diff
@@ -779,13 +805,28 @@ private ObjectDefinition _updateObjectDefinition(
 		objectDefinition.setPluralLabelMap(pluralLabelMap);
 
 		if (objectDefinition.isApproved()) {
+			List<ObjectEntry> objectEntries =
+				_objectEntryLocalService.getObjectEntries(
+					0, objectDefinition.getObjectDefinitionId(),
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java]: We must get the groupId from ObjectScopeProvider

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 3]
```diff
@@ -779,13 +805,28 @@ private ObjectDefinition _updateObjectDefinition(
...
+
 				objectDefinitionLocalService.undeployObjectDefinition(
 					objectDefinition);
 			}
 			else if (active) {
 				objectDefinitionLocalService.deployObjectDefinition(
 					objectDefinition);
+
+				for (ObjectEntry objectEntry : objectEntries) {
+					_objectEntryLocalService.updateObjectEntry(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java]: I think we should create a method to just call WorkflowHandlerRegistryUtil.startWorkflowInstance, what do you think?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 4]
```diff
@@ -945,6 +986,9 @@ private void _validateVersion(boolean system, int version)
 
 	private BundleContext _bundleContext;
 
+	@Reference
+	private CompanyLocalService _companyLocalService;
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java]: This service is not being used, please remove it.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 5]
```diff
@@ -666,6 +672,26 @@ private void _createTable(
...
+
+		List<ObjectEntry> objectEntriesPending = stream.filter(
+			listTypeEntry -> listTypeEntry.getStatus() == 1
+		).collect(
+			Collectors.toList()
+		);
+
+		for (ObjectEntry objectEntry : objectEntriesPending) {
+			_workflowInstanceLinkLocalService.deleteWorkflowInstanceLink(
+				objectEntry.getCompanyId(), objectEntry.getCompanyId(),
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java]: You are passing CompanyId as GroupId, please fix this.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by guilhermedcamacho** [comment]
Resent on #143

**Comment by mateussgn** [comment]
@guilhermedcamacho I have seen your comments, thanks for reviewing this pull request, made de necessary changes and forwarded to Bchan.

**Pull request closed without merging**